### PR TITLE
Add ESP32

### PR DESCRIPTION
**Pull request by Krupskis**
This is code and documentation for esp32c3, not tested with other esp32 versions, but should work. https://www.seeedstudio.com/Seeed-XIAO-ESP32C3-p-5431.html

The whole esp32 setup now is around $20 with battery.

### CODE REVIEW

File: devices/esp32-c3/main/build/log/idf_py_stderr_output_4114 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+CMake Warning (dev) in CMakeLists.txt:
```
Comment (adamcohenhillel) [inline devices/esp32-c3/main/build/log/idf_py_stderr_output_4114:1]: lets add /log to gitignore?

File: devices/esp32-c3/main/build/CMakeFiles/CMakeConfigureLog.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (adamcohenhillel) [inline devices/esp32-c3/main/build/CMakeFiles/CMakeConfigureLog.yaml:1]: same here, lets add to gitignore

File: devices/esp32-c3/main/recorder/I2SRecorder.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#include "I2SRecorder.h"
```
[the remark this answers is not in the document]
Comment (Krupskis) [reply devices/esp32-c3/main/recorder/I2SRecorder.cpp:1]: Yeah that could work

File: devices/esp32-c3/main/build/log/idf_py_stderr_output_4114 (hunk 1 above)
Comment (Krupskis) [reply devices/esp32-c3/main/build/log/idf_py_stderr_output_4114:1]: I deleted the whole build directory and added that to .gitignore

File: devices/esp32-c3/main/build/CMakeFiles/CMakeConfigureLog.yaml (hunk 2 above)
Comment (Krupskis) [reply devices/esp32-c3/main/build/CMakeFiles/CMakeConfigureLog.yaml]: I deleted the whole build directory and added that to .gitignore

**Comment by Krupskis** [comment]
@adamcohenhillel this is ready to be merged

**Review by adamcohenhillel — APPROVED** [review]

**Pull request merged**